Xamarin.Forms.Core/CarouselView/CarouselView.cs

```diff
@@ -0,0 +1,133 @@
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+IN THE SOFTWARE.
+ */
+
+namespace Xamarin.Forms
+{
+	/// <summary>
+	/// CarouselView Interface
+	/// </summary>
+	public class CarouselView : View
+	{
+		public static readonly BindableProperty OrientationProperty = BindableProperty.Create("Orientation", typeof(CarouselViewOrientation), typeof(CarouselView), CarouselViewOrientation.Horizontal);
+
+		public CarouselViewOrientation Orientation
+		{
+			get { return (CarouselViewOrientation)GetValue(OrientationProperty); }
+			set { SetValue(OrientationProperty, value); }
+		}
+
+		// Android and iOS only
+		public static readonly BindableProperty InterPageSpacingProperty = BindableProperty.Create("InterPageSpacing", typeof(int), typeof(CarouselView), 0);
+
+		public int InterPageSpacing
+		{
+			get { return (int)GetValue(InterPageSpacingProperty); }
+			set { SetValue(InterPageSpacingProperty, value); }
+		}
+
+		public static readonly BindableProperty IsSwipingEnabledProperty = BindableProperty.Create("IsSwipingEnabled", typeof(bool), typeof(CarouselView), true);
+
+		public bool IsSwipingEnabled
+		{
+			get { return (bool)GetValue(IsSwipingEnabledProperty); }
+			set { SetValue(IsSwipingEnabledProperty, value); }
+		}
+
+		public static readonly BindableProperty IndicatorsTintColorProperty = BindableProperty.Create("IndicatorsTintColor", typeof(Color), typeof(CarouselView), Color.Silver);
+
+		public Color IndicatorsTintColor
+		{
+			get { return (Color)GetValue(IndicatorsTintColorProperty); }
+			set { SetValue(IndicatorsTintColorProperty, value); }
+		}
+
+		public static readonly BindableProperty CurrentPageIndicatorTintColorProperty = BindableProperty.Create("CurrentPageIndicatorTintColor", typeof(Color), typeof(CarouselView), Color.Gray);
+
+		public Color CurrentPageIndicatorTintColor
+		{
+			get { return (Color)GetValue(CurrentPageIndicatorTintColorProperty); }
+			set { SetValue(CurrentPageIndicatorTintColorProperty, value); }
+		}
+
+		public static readonly BindableProperty IndicatorsShapeProperty = BindableProperty.Create("IndicatorsShape", typeof(IndicatorsShape), typeof(CarouselView), IndicatorsShape.Circle);
+
+		public IndicatorsShape IndicatorsShape
+		{
+			get { return (IndicatorsShape)GetValue(IndicatorsShapeProperty); }
+			set { SetValue(IndicatorsShapeProperty, value); }
+		}
+
+		public static readonly BindableProperty ShowIndicatorsProperty = BindableProperty.Create("ShowIndicators", typeof(bool), typeof(CarouselView), false);
+
+		public bool ShowIndicators
+		{
+			get { return (bool)GetValue(ShowIndicatorsProperty); }
+			set { SetValue(ShowIndicatorsProperty, value); }
+		}
+
+		public static readonly BindableProperty ItemsSourceProperty = BindableProperty.Create("ItemsSource", typeof(IEnumerable), typeof(CarouselView), null);
```

Comment:
source and templating should be handled in the Core, reusing code from listview

Reply:
?